src/pages/main/main-page.tsx

```diff
@@ -1,17 +1,45 @@
+import { useState } from 'react';
+import { useAppSelector } from '../../hooks/redux';
 import AllCards from '../../components/allCards/allCards';
-import { CardData } from '../../constants/cardsData/cardsData';
-import React from 'react';
 import Search from '../../components/search/search';
+import Modal from '../../components/modal/modal';
+import Loader from '../../components/loader/loader';
+import ErrorMessage from '../../components/errorMessage/errorMessage';
+import { useGetDataQuery } from '../../store/reducers/apiSlice';
+import classes from './main.module.scss';
 
-class Main extends React.Component {
-  render() {
-    return (
-      <section>
-        <Search />
-        <AllCards cards={CardData} />
-      </section>
-    );
-  }
-}
+const textError = {
+  notFound: 'Nothing found',
+};
+
+const Main = () => {
+  const { textSearch } = useAppSelector((state) => state.searchReducer);
+  const [idD, setId] = useState<number>(0);
+  const [showModal, setShowModalt] = useState(false);
+  const { data, isFetching, isError } = useGetDataQuery(textSearch);
+
+  const onShowModal = (id: number) => {
+    setId(id);
+    setShowModalt(!showModal);
+  };
+
+  const onShowModal2 = () => {
+    setShowModalt(!showModal);
+  };
+
+  return (
+    <section className={classes.mainSection}>
+      <Search />
+      {isFetching ? (
+        <Loader />
+      ) : isError ? (
```

Comment:
it is very complicated to read:
```
{isFetching && <Loader />}
{!isFetching && isError && <ErrorMessage message={textError.notFound} />}
{!isFetching && !isError && <AllCards onShowModal={onShowModal} results={data.results} />}
```
I think it will be a little bit more readable)